### PR TITLE
crosscluster/logical: cancel producer job when consumer cancels 

### PR DESCRIPTION
**Pull request by stevendanna**
Previously, we were leaving the producer job in place when the
consumer job was canceled. The producer job would time out after 24
hours. However, until then, the PTS record on the source would remain.

In some ways this is nice as it allows the user to resume the stream
from a cursor in the case where they have the need to cancel and
resume the job for some reason.  However, our goal is that there are
no such cases (the user should simply need to PAUSE/ALTER/RESUME in
such cases).

Release note: None
Epic: none

### CODE REVIEW

File: pkg/ccl/crosscluster/logical/logical_replication_job.go [hunk 1]
```diff
@@ -588,20 +591,56 @@ func loadOnlineReplicatedTime(
...
+			}
+			defer closeAndLog(ctx, client)
+			return client.Complete(ctx, streamID, false /* successfulIngestion */)
+		},
+	); err != nil {
+		log.Warningf(ctx, "error completing the source cluster producer job %d: %s", streamID, err.Error())
+	}
+}
+
+func closeAndLog(ctx context.Context, d streamclient.Dialer) {
```
Comment (navsetlur) [inline pkg/ccl/crosscluster/logical/logical_replication_job.go:638]: In the ingestion job we do streamclient.Client.close(). Why use the type Dialer here?

**Review by navsetlur — APPROVED** [review]
LGTM

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go [hunk 2]
```diff
@@ -278,18 +278,16 @@ func TestLogicalStreamIngestionJob(t *testing.T) {
 	dbA.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", dbBURL.String()).Scan(&jobAID)
 	dbB.QueryRow(t, "CREATE LOGICAL REPLICATION STREAM FROM TABLE tab ON $1 INTO TABLE tab", dbAURL.String()).Scan(&jobBID)
 
-	now := server.Server(0).Clock().Now()
-	t.Logf("waiting for replication job %d", jobAID)
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:282]: Well, it'll still be in the test output. But now if you don't want it there, you can remove one more line :D.

File: pkg/ccl/crosscluster/logical/logical_replication_job.go (hunk 1 above)
Comment (stevendanna) [reply pkg/ccl/crosscluster/logical/logical_replication_job.go:638]: The most honest answer is this code was copy/pasta from the PCR job.

If I try to recall why we wrote that code in this way, I think it was mostly just that we want to log the error so we put it in the function and then we used streamclient.Dialer so we could use it in multiple places that have different kinds of clients.

File: pkg/ccl/crosscluster/logical/logical_replication_job_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (navsetlur) [reply pkg/ccl/crosscluster/logical/logical_replication_job_test.go:282]: ah I didn't see you moved it to the other function

**Comment by stevendanna** [comment]
TFTR+

bors r+

**Pull request merged**